### PR TITLE
Switch to poetry

### PR DESCRIPTION
**Pull request by dpassen**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[tool.poetry]
+name = "optional.py"
+version = "1.1.0"
+description = "An implementation of the Optional object in Python"
+
+license = "MIT"
+
+authors = [
+    "Chad Befus <crbefus@gmail.com>"
```
Comment (cbefus) [inline pyproject.toml]: This takes a list, so we should add your name here too!

**Comment by dpassen** [comment]
I don't remember if we ever attempted to publish this branch to Test PyPI. I had to kill Python 3.4 too =\

**Comment by cbefus** [comment]
> I don't remember if we ever attempted to publish this branch to Test PyPI. I had to kill Python 3.4 too =\

Im OK with killing off 3.4

File: test_requirements.txt [hunk 2]
```diff
@@ -1,11 +0,0 @@
-certifi==2018.8.24
```
Comment (cbefus) [inline test_requirements.txt]: If we remove this file we need to update the `README.md` for how to install test requirements.

**Comment by cbefus** [comment]
@dpassen  So I pulled this and ran the tests.  Its really nice.  I made 2 comments on this review.  Adding your name to the Authors is up to you but we need to update the `README.md` for running tests.
```
pip install poetry
poetry install
poetry run pytest --cov=optional
```

**Comment by dpassen** [comment]
@cbefus Found some time to do both

**Comment by dpassen** [comment]
@cbefus bump

**Pull request merged**